modules/config/keyboard.nix

```diff
@@ -0,0 +1,103 @@
+{ mode, config, lib, ... }:
+
+with lib;
+let
+  cfg = config.soxin.settings.keyboard;
+
+  layoutModule = types.submodule ({ config, ... }: {
+    options = {
+      layout = mkOption {
+        type = types.str;
+        default = "us";
```

Comment:
If you pick the config of NixOS here as the default, then it should help people migrate over.

Reply:
The problem with using NixOS (or home-manager) configuration as a default here is that they both are comma-separated strings (for layout and variant). Iterating over both of those and settings our config (list of attributes) would be too much of a pain, and I don't think it's asking a lot from people who are migrating to setup their keyboard configuration.